pyqtgraph/__init__.py

```diff
@@ -463,3 +464,14 @@ def stack(*args, **kwds):
     except NameError:
         consoles = [c]
     return c
+
+
+def applyPalette(app,name):
```

Comment:
Needs documentation before merge, but we can work that out when we know that this is the way we want to do things.
(I like it.)

Reply:
I'm changing the `name` parameter to `style`. Do you think we should set a default where `style = qarkstyleDark`?

Reply:
I think we want people to read the docs on this, so I think it is better not to set a default.
We should accept a QPalette for the `style` parameter, though, and we should check the base color to update the `darkMode` flag!

Reply:
>  We should accept a QPalette for the style parameter, though, and we should check the base color to update the darkMode flag!

Do you mean in addition to accepting a string like this?
```py
def applyPalette(app, style):
    if isinstance(style, str):
        style = style.lower()
        if style in ['qdarkstylelight', 'light']:
            p = palette.getQDarkStyleLightQPalette()
        elif style in ['qdarkstyle', 'dark']:
            p = palette.getQDarkStyleDarkQPalette()
        else:
            raise ValueError(f'no palette by the name {style} exists')
    elif isinstance(style, QtGui.QPalette):
        p = style
        app.setPalette(p)
    else:
        raise TypeError('style either be a string or QPalette')
    app.setPalette(p)
 ```

Reply:
Yes. Except for not setting the palette twice, maybe.

And with the overall addition of the basecolor lightness check from mkQApp.

Reply:
> Yes. Except for not setting the palette twice, maybe.

I just wanted to make sure it was really set. Like really set... 
Fixed!

>
And with the overall addition of the basecolor lightness check from mkQApp.
That would go just before or after the final setPalette call, when we know that p contains a QPalette.

Gotcha will emit the palette through paletteChanged, though... is the darkMode property used other than the example app?

Reply:
Ah, if we catch the change there and set the flag, then we're all good!

Let's assume for now that *some* part of the code *somewhere* probably does *something* with the darkMode flag. So it's nice to always have it up-to-date.